dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/resolver/CategoryOptionGroupResolver.java

```diff
@@ -97,17 +97,16 @@ private String getExpression( String expression, DimensionalItemId id, List<Stri
     {
         List<String> cocUidIntersection = getCategoryOptionCombosIntersection( cogUidList, dataElementId );
 
-        if ( isEmpty( cocUidIntersection ) )
+        if ( Optional.ofNullable( cocUidIntersection ).filter( i -> !i.isEmpty() ).isPresent() )
```

Comment:
Empty strings will still be in the `resolved` collection below in case there are also non-empty strings in `cocUidIntersection`. I wonder if this filtering should not rather move into `getCategoryOptionCombosIntersection` instead.